processing/src/main/java/io/druid/query/search/SearchQueryRunner.java

```diff
@@ -183,7 +186,26 @@ public void updateSearchResultSet(
       }
     }
   }
-  
+
+  public static class DoubleSearchColumnSelectorStrategy implements SearchColumnSelectorStrategy<DoubleColumnSelector>
+  {
+    @Override
+    public void updateSearchResultSet(
+        String outputName,
+        DoubleColumnSelector dimSelector,
+        SearchQuerySpec searchQuerySpec,
+        int limit,
+        Object2IntRBTreeMap<SearchHit> set
+    )
+    {
+      if (dimSelector != null) {
+        final String dimVal = String.valueOf(dimSelector.get());
```

Comment:
There could be a single implementation for DoubleColumnSelector, FloatColumnSelector and LongColumnSelector, that has `Function<ValueSelectorType, String> getAsString` field.